config/routes.rb

```diff
@@ -11,6 +11,8 @@
       # OPTIONS action for each collection
       match collection_name.to_s, :controller => collection_name, :action => :options, :via => :options, :as => nil
 
+      collection_name_plural, collection_name_singular = Api::Utils.generate_plural_and_singular_alias(collection_name.to_s)
```

Comment:
can't we always generate the unique singular ?

```
collection_alias = collection_name.to_s.pluralize
resource_alias   = #{collection_name.to_s.singularize}_resource
```

using collection_alias and resource_alias instead of collection_name_plural and collection_name_singular.

We won't need the added method in Api::Utils and simplify this PR a bit.

Reply:
But I agree with you on the variables naming. I'm going to change them to collection_alias and resource_alias.